aslam_offline_calibration/kalibr/python/kalibr_common/ImageDatasetReader.py

```diff
@@ -129,6 +129,10 @@ def getImage(self, idx):
         self.uncompress = uncompress
       img_data = np.reshape(self.uncompress(np.fromstring(
           data.data, dtype='uint8')), (data.height, data.width), order="C")
+    elif data._type == 'sensor_msgs/CompressedImage':
+      img_data = np.array(self.CVB.compressed_imgmsg_to_cv2(data))
+      if len(img_data.shape) > 2 and img_data.shape[2] == 3:
+        img_data = cv2.cvtColor(img_data, cv2.COLOR_RGB2GRAY)
```

Comment:
I guess we should check for the encoding type here also (see below).

Reply:
It seems compressed images are typically only BGR and mono. The encoding isn't stored in the message type, so even if it was encoded differently, it seems that this can't be found out besides inspecting the data type?
http://docs.ros.org/en/noetic/api/sensor_msgs/html/msg/CompressedImage.html
https://github.com/ros-perception/vision_opencv/blob/906d326c146bd1c6fbccc4cd1268253890ac6e1c/cv_bridge/src/cv_bridge.cpp#L480-L506